superagi/tools/file/append_file.py

```diff
@@ -42,12 +46,23 @@ def _execute(self, file_name: str, content: str):
         Returns:
             success or error message.
         """
+        
         final_path = ResourceHelper.get_agent_write_resource_path(file_name, Agent.get_agent_from_id(
             session=self.toolkit_config.session,
             agent_id=self.agent_id),
           AgentExecution.get_agent_execution_from_id(
               session=self.toolkit_config.session,
               agent_execution_id=self.agent_execution_id))
+        
+        if StorageType.get_storage_type(get_config("STORAGE_TYPE", StorageType.FILE.value)) == StorageType.S3:
+            previous_content = self.get_previous_content(final_path)
+            try:
```

Comment:
lets say it is not a txt file and the append tool is used wont is delete the file, even though the operation is not supported